x-pack/platform/plugins/shared/cloud/server/plugin.ts

```diff
@@ -265,6 +265,10 @@ export class CloudPlugin implements Plugin<CloudSetup, CloudStart> {
           : undefined;
 
         const solutionType = this.config.onboarding?.default_solution;
+        if (queryOnboardingToken) {
+          normalizedRoute.searchParams.set('onboarding_token', queryOnboardingToken);
```

Comment:
Why did you add this vs reading the token from the saved object that was added for us to read this value? 

See the next block of code that persists this token to a saved object so that we can read after it's no longer included in query parameters.

Reply:
I started with that approach, but that was actually much more complicated--I had to move a lot of code around to get the types in the right place, had to add a backend route to retrieve that token on every load, that backend route needed a savedObjectClient with the right parameters--and that parameter constant was only accessible within the cloud server, so I had to move that around as well. And the fact that it was an async call to retrieve the token made populating it initially more complicated (now we'd need loading states).

Basically: I ended up with a very complicated and messy pull request that also felt very brittle. This felt much more straightforward.

Reply:
Let me give grabbing it from the saved object another go

Reply:
Updated to use onboarding token in a cleaner way than before